common/cluster/archival_status.go

```diff
@@ -32,8 +32,9 @@ type (
 	// ArchivalConfig is an immutable representation of the current archival configuration of the cluster
 	ArchivalConfig struct {
 		status                 ArchivalStatus
-		defaultBucket          string
 		enableReadFromArchival bool
+		domainDefaultStatus    ArchivalStatus
```

Comment:
Looks like you do not have getters for these methods?

Reply:
I think now that we have two types of entities in this archivalConfig: domainDefaults and clusterDefaults the getters for this file might look a little weird. Lets revisit what these getters should look like when we make domain CRUD changes to workflowHandler.go

Reply:
One option is to introduce two new structs clusterConfig and domainDefaultConfig. Then just have getters to these two methods and have these structs themselves have the needed methods on them.

Reply:
Will improve this part when modifying frontend domain handler as it's not clear now how these fields will be used.